resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientRequestHeaders.java

```diff
@@ -293,9 +308,13 @@ public MediaType getMediaType()
          }
          else
          {
-            String str = configuration.toHeaderString(obj);
-            Cookie cookie = Cookie.valueOf(str);
-            cookies.put(cookie.getName(), cookie);
+            String[] cs = ((String) obj).split(";");
+            for (String c : cs)
+            {
+               String str = configuration.toHeaderString(c);

```

Comment:
re: "BTW, why not use CookieParser?"

Well, yeah. Cookie.valueOf() uses CookieHeaderDelegate, which I've fixed now. I think.